### PR TITLE
Feature/623 offset to opex defaults to organization

### PR DESCRIPTION
**Pull request by Elaine-Krauss-TCG**
#623

### CODE REVIEW

File: front-end/src/app/shared/models/transaction-types/OFFSET_TO_OPERATING_EXPENDITURES.model.ts [hunk 1]
```diff
@@ -13,13 +13,15 @@ import {
...
   TransactionNavigationControls,
 } from '../transaction-navigation-controls.model';
+import { ContactTypes } from '../contact.model';
 
 export class OFFSET_TO_OPERATING_EXPENDITURES implements TransactionType {
   scheduleId = 'A';
   componentGroupId = 'B';
   isDependentChild = false;
   title = LabelUtils.get(ScheduleATransactionTypeLabels, ScheduleATransactionTypes.OFFSET_TO_OPERATING_EXPENDITURES);
   schema = schema;
```
Comment (toddlees) [inline front-end/src/app/shared/models/transaction-types/OFFSET_TO_OPERATING_EXPENDITURES.model.ts:23]: This is a great direction to go!  I'm a big fan of having transactionType-level configurations in the transactionType classes

File: front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.ts [hunk 2]
```diff
@@ -63,6 +63,11 @@ export abstract class TransactionTypeBaseComponent implements OnInit, OnDestroy
     validateService.formValidatorSchema = transactionType?.schema;
     validateService.formValidatorForm = form;
 
+    // Override contact type options if present in transactionType
+    if (transactionType && transactionType.contactTypeOptions) {
+      this.contactTypeOptions = this.getContactTypeOptions(transactionType.contactTypeOptions);
```
Comment (toddlees) [inline front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.ts]: is this essentially equivalent to:
```
    if (transactionType && transactionType.contactTypeOptions) {
      this.contactTypeOptions = transactionType.contactTypeOptions.map((type: ContactTypes) => ({
        name: LabelUtils.get(ContactTypeLabels, type),
        code: type,
      }));
    }

Comment (toddlees) [reply front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.ts]: I could see us using this pattern elsewhere when we want a PrimeOptions list given a list of 'code's and one of our homebrew LabelLists

maybe in LabelUtils we could add

```

  public static getPrimeOptions(labelArrays: LabelList, options: string[]) {
    return options.map((type: string) => ({
      name: LabelUtils.get(labelArrays, type),
      code: type,
    }));
  }

Comment (mjtravers) [reply front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.ts]: I agree that the getContactTypeOptions() code would be good to generalize out into the LabelUtils class.

Comment (Elaine-Krauss-TCG) [reply front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.ts]: I like this idea a lot.  I'll take a look at it!

Comment (Elaine-Krauss-TCG) [reply front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.ts]: https://github.com/fecgov/fecfile-web-app/pull/813/files#diff-5bf0fd4e8831d3a8ff3905e7c9a0a450bbeb72fb77c6cd9c7438a604a6925517R55

LabelUtil's getPrimeOptions was updated to allow you to optionally specify the values to use.

**Review by toddlees — APPROVED** [review]

**Review by dheitzer — APPROVED** [review]

**Review by mjtravers — APPROVED** [review]

**Pull request merged**